src/relay/backend/vm/compiler.cc

```diff
@@ -924,19 +924,20 @@ IRModule VMCompiler::OptimizeModule(const IRModule& mod, const TargetsMap& targe
   pass_seqs.push_back(transform::LambdaLift());
   pass_seqs.push_back(transform::InlinePrimitives());
 
+  // Manifest the allocations.
+  pass_seqs.push_back(transform::ManifestAlloc(this->target_host_));
+  // Compute away possibly introduced constant computation.
+  pass_seqs.push_back(transform::FoldConstant());
+  // Fuse the shape functions.
+  pass_seqs.push_back(transform::FuseOps());
+
```

Comment:
Why is this change for?

Reply:
zhiics previously made a PR to outline all partition function to avoid being touched by some passes such as `fuse_ops`, but you can see that before this PR the second run of `fuse_ops` pass is invoked after the inline pass, resulting in the decomposition of `batch_norm`. After this change, we inline the partition functions back after finishing all other passes so that `batch_norm` could be preserved safely to external codegen.

Reply:
Yeah, inlining should be after fusion. I missed the second fusion...

Reply:
ok makes sense, but batch_norm is decomposed by SimplifyInference, fuse_ops does something else.

Reply:
yeah, if batch_norm is encapsulated in an external function, it will be lifted out and therefore invisible by simplifyinference. Later on, if we inline it and run fusion, fusion pass will see this op and complain that on OpPattern is registered for this op...